### DISCUSSION TITLE
Extending Modern Search Web Parts: Customizing Search Box and Verticals

### DISCUSSION THREAD

**Discussion by SAgnihotri200895** [microsoft-search/pnp-modern-search#3906] [Ideas]
### Background
I'm working with the PnP Modern Search solution (v4) and I'm trying to customize the appearance of the search box and search verticals web parts. The documentation mentions different levels of customization:

- 'Basic' customizations (HTML, CSS, Handlebars)
- 'Advanced' customizations (SharePoint Framework development)

However, I'm encountering some confusion regarding the extensibility possibilities for these specific web parts.

### Current Understanding
According to the [extensibility documentation](https://microsoft-search.github.io/pnp-modern-search/extensibility/):

Search Box Web Part supports custom suggestions providers.
Search Verticals Web Part is listed as supporting no extensions.

### My Requirements

I want to:

- Customize the appearance of the search box (add background, change layout).
- Modify the layout of the search verticals.

**Questions**

- Is it possible to customize the appearance of the search box beyond just creating a custom suggestions provider? If so, how?
- Despite the documentation stating no supported extensions for Search Verticals, is there any way to customize their appearance or layout?
- If these customizations are possible, do they fall under 'Basic' customizations (achievable with HTML, CSS, and Handlebars), or do they require 'Advanced' customizations with SharePoint Framework development?
- Are there any examples or resources available for customizing these specific web parts?

Any guidance, examples, or clarification on the extensibility possibilities for these web parts would be greatly appreciated. Thank you!

**Comment by wobba** [comment]
If you want to change appearance of the search box you likely need/want to change this in the code itself and build a custom variant. The better approach is to extend the web part with properties to control the layout more, and then PR this as a change for everyone to benefit from. 

It could be for example to provide a template option for the search box web part, allowing you to enter html/css there around the search box control itself - and having properties on the control to allow changes.
